src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsServiceTest.java

```diff
@@ -19,19 +34,47 @@
 @Tag(Unit.TAG)
 @EmbeddedDatabaseTest
 public class SnapshotBuilderSettingsServiceTest {
+  private AuthenticatedUserRequest testUser = AuthenticationFixtures.randomUserRequest();
+
   @Mock private SnapshotBuilderSettingsDao snapshotBuilderSettingsDao;
+  @Mock private AzureSynapsePdao azureSynapsePdao;
+  @Mock private AzureSynapseService azureSynapseService;
+  @Mock private DatasetDao datasetDao;
 
   private SnapshotBuilderService snapshotBuilderService;
 
   @BeforeEach
   public void setUp() {
-    snapshotBuilderService = new SnapshotBuilderService(snapshotBuilderSettingsDao);
+    snapshotBuilderService =
+        new SnapshotBuilderService(
+            snapshotBuilderSettingsDao, azureSynapsePdao, azureSynapseService, datasetDao);
   }
 
   @Test
   public void testGetSnapshotBuilderSettings() {
     UUID datasetId = UUID.randomUUID();
-    snapshotBuilderService.getSnapshotBuilderSettings(datasetId);
+    when(snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(datasetId))
+        .thenReturn(SnapshotBuilderTestData.SETTINGS);
+    when(azureSynapsePdao.getStatsForIntColumn(any(), any(), any(), any()))
+        .thenReturn(new ColumnStatisticsIntModel().minValue(0).maxValue(10));
+    when(datasetDao.retrieve(datasetId))
+        .thenReturn(
+            new Dataset(new DatasetSummary().cloudPlatform(CloudPlatform.AZURE))
```

Comment:
I thought about pulling this off into `SnapshotBuilderTestData`. Any thoughts or preferences on that?